packages/core/__tests__/middleware.js

```diff
@@ -108,7 +111,7 @@ test('enhance channel.put with an emitter', () => {
   store.dispatch({ type: 'e' })
 
   // saga must be able to take actions emitted by middleware's custom emitter
-  const expected = ['a', 'b', 'c', 'd', 'e']
+  const expected = ['a', 'put_a', 'b', 'put_b', 'c', 'put_c', 'd', 'put_d', 'e', 'put_e']
```

Comment:
😕  I think the expected result should be `['a', 'b', 'c', 'd', 'e']`, because we are calling `actual.push(ac.type)` only when `ac.type.length === 1`.

If I change the `saga` into...
```javascript
function* saga() {
  yield takeEvery('*', ac => actual.push(ac.type))
  yield takeEvery(ac => ac.type.length === 1, function*(ac) {
    yield put({ type: `put_${ac.type}` })
  })
}
```
..., then the expected result should be `['a', 'put_a', 'b', 'put_b', 'c', 'put_c', 'd', 'put_d', 'e', 'put_e']`.